jupyterhub/user.py

```diff
@@ -568,7 +568,7 @@ async def spawn(self, server_name='', options=None, handler=None):
                 client_id,
                 api_token,
                 url_path_join(self.url, server_name, 'oauth_callback'),
-                description="Server at %s"
+                description="Server at %s"  # todo: own server: 'users:servers!user={username} as allowed role?
```

Comment:
I think the todo here is make server allowed_roles configurable.

Logical question: should this be a Spawner configuration or an Authenticator configuration? This is an open question!

Reply:
For now, as discussed, I'll go for adding the roles on a Spawner configuration, then we can evaluate the setup later